pkg/iter/sc_finder.go

```diff
@@ -127,7 +127,13 @@ func (m *SubclusterFinder) FindSubclusters(ctx context.Context, flags FindFlags,
 		// indication the subcluster is being removed.
 		for i := range missingSts.Items {
 			scName := missingSts.Items[i].Labels[vmeta.SubclusterNameLabel]
-			subclusters = append(subclusters, &vapi.Subcluster{Name: scName, Size: 0})
+			subclusters = append(subclusters, &vapi.Subcluster{
+				Name: scName,
+				Size: 0,
+				Annotations: map[string]string{
+					vmeta.StsNameOverrideAnnotation: missingSts.Items[i].ObjectMeta.Name,
```

Comment:
Why do you need to set this when the subcluster is being removed?

Reply:
It's because the statefulset still exists and callers need to be able to generate its name to delete it. Consider the scenario where the VerticaDB has been updated by removing a subcluster but we haven't finished removing it in the Vertica catalog. We need to do the cleanup logic, which includes eventually deleting the statefulset itself.